packages/salesforcedx-sobjects-faux-generator/src/generator/fauxClassGenerator.ts

```diff
@@ -25,6 +25,39 @@ export class FauxClassGenerator {
   private STANDARDOBJECTS_DIR = 'standardObjects';
   private CUSTOMOBJECTS_DIR = 'customObjects';
 
+  // the empty string is used to represent the need for a special case
+  // usually multiple fields with specialized names
+  private static typeMapping: Map<string, string> = new Map([
+    ['string', 'String'],
+    ['double', 'Decimal'],
+    ['reference', ''],
+    ['boolean', 'Boolean'],
+    ['currency', 'Currency'],
+    ['date', 'Date'],
+    ['datetime', 'Datetime'],
+    ['email', 'Email'],
+    ['location', 'Location'],
+    ['percent', 'Decimal'],
+    ['phone', 'String'],
+    ['picklist', 'String'],
+    ['multipicklist', 'String'],
+    ['textarea', 'String'],
+    ['encryptedstring', 'String'],
+    ['url', 'String'],
+    ['id', 'Id'],
+    // note that the mappings below "id" only occur in standard SObjects
+    ['base64', 'Blob'],
+    ['address', 'Address'],
+    ['int', 'Integer'],
+    ['anyType', 'Object'],
+    ['combobox', 'String'],
+    ['time', 'Time'],
+    // TBD what are these mapped to and how to create them
+    //['calculated', 'xxx'],
```

Comment:
Calculated is for formula fields, where the type can be vary depending on what kind of formula it is. If you just get the type of the field, it should be limited to checkbox, currency, date, datetime, number, percent, text, and time. These are already mapped above so this should be safe to ignore.

Reply:
agreed - I tried a number of different formula fields and they gave the appropriate results - not sure whether calculated was used at one time and not any more or whether there is something out there somewhere that still uses it - but I couldn't find anything

Reply:
This can be taken out. The types of the calculated field are known.

Reply:
Yes, but, both of these exist as legitimate describe field types, but I haven't been able to figure out how to produce them and what they should map to (if they can be produced at all).   Feels wrong to just leave them out when we know that they are missing.   Next person will have no clue if they ever show up